clients/src/main/java/org/apache/kafka/common/requests/FetchRequest.java

```diff
@@ -302,6 +315,18 @@ public String toString() {
         }
     }
 
+    public static void updateFetchRequestDataReplicaState(FetchRequestData fetchRequestData, int replicaId, long replicaEpoch, short version) {
+        if (version < 15) {
+            fetchRequestData.setReplicaId(replicaId);
+            fetchRequestData.setReplicaState(new ReplicaState());
+        } else {
+            fetchRequestData.setReplicaId(new FetchRequestData().replicaId());
```

Comment:
nit: Allocating `FetchRequestData` to get `-1` is a bit wasteful here. Could we just use `-1`?

Reply:
Actually, the default value is 0, I can make a change to have it -1.

Reply:
Changed to -1.